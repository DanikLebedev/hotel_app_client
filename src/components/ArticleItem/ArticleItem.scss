.articles-page__item-img {
  object-fit: cover;
  width: 300px;
  height: 300px
}