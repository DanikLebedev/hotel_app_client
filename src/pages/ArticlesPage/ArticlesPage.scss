.articles-page {
  &-bg {
    background: url("/images/543280.jpg") center center / cover;
    height: 50vh;
    width: 100%;
    position: absolute;
    top: 0;
    left: 0;
    z-index: -3;
  }
  h1 {
    color: #ffffff;
    font-size: 4em;
  }
  &-wrapper {
    padding-top: 35vh;
  }
}

@media screen
and (min-width: 768px)
and (max-width: 1199px){
  .articles-page {
    h1 {
      font-size: 3em;
    }
    &-wrapper {
      padding-top: 40vh;
    }
  }
}


@media screen and (min-width: 300px) and (max-width: 480px) {
  .articles-page {
    h1 {
      font-size: 2em;
    }
    &-wrapper {
      padding-top: 35vh;
    }
  }
}
