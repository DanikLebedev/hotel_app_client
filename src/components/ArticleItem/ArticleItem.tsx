import React from 'react';
import { config } from '../../config';
import { Article, Room } from '../../interfaces/clientInterfaces';
import { Col, Container, Row } from 'react-bootstrap';
import './ArticleItem.scss';
import { NavLink } from 'react-router-dom';

interface ArticleItemProps {
    articleInfo: Article;
}

function kitcut(text, limit) {
    text = text.trim();
    if (text.length <= limit) return text;

    text = text.slice(0, limit);
    const lastSpace = text.lastIndexOf(' ');

    if (lastSpace > 0) {
        text = text.substr(0, lastSpace);
    }
    return text + '...';
}

const ArticleItem: ({ articleInfo }: ArticleItemProps) => any = ({ articleInfo }: ArticleItemProps) => {
    return (
        <Col lg={4} md={4} sm={4} xs={4} className={'mb-4'} >
            <img className="articles-page__item-img" src={config.baseUrl + articleInfo.image} alt="img" />
            <h3>{articleInfo.title}</h3>
            <div className="rooms-page__item-text">
                <p>{articleInfo.createdAt ? new Date(articleInfo.createdAt).toLocaleDateString(): null}</p>
                <p>{kitcut(articleInfo.text, 100)}</p>
            </div>
            <NavLink to={`/articles/${articleInfo._id}`} id={`redirect-button`} className="button btn-black">
                Read More
            </NavLink>
        </Col>
    );
};

export default ArticleItem;
