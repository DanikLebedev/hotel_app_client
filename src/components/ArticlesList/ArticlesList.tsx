import React, { useContext } from 'react';
import RoomItem from '../RoomItem/RoomItem';
import Loader from '../Loader/Loader';
import { Article, Room } from '../../interfaces/clientInterfaces';
import { ClientContext } from '../../context/client.context';
import ArticleItem from '../ArticleItem/ArticleItem';

export const ArticlesList: React.FC = (): JSX.Element => {
    const fetchedArticles: Article[] = useContext(ClientContext).fetchedAllArticles;
    return (
        <div>
            {!fetchedArticles ? (
                <div className="d-flex justify-content-center align-items-center vh-100">
                    <Loader />
                </div>
            ) : (
                fetchedArticles.map((article: Article, i: number) => {
                    return <ArticleItem key={article.title + i} articleInfo={article} />;
                })
            )}
        </div>
    );
};

export default ArticlesList;
