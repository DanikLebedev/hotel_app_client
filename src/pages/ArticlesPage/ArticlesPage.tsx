import React, { FC } from 'react';
import { Container } from 'react-bootstrap';
import './ArticlesPage.scss';
import FindRoomForm from '../../components/FindRoomForm/FindRoomForm';
import ArticlesList from '../../components/ArticlesList/ArticlesList';
import {withTranslation} from "react-i18next";

const ArticlesPage: FC = ({t}:any): JSX.Element => {
    return (
        <div className="articles-page">
            <FindRoomForm />
            <div className="articles-page-bg d-flex justify-content-center align-items-end">
                <h1>{t('news.label')}</h1>
            </div>
            <Container className="articles-page-wrapper">
                <ArticlesList />
            </Container>
        </div>
    );
};

export default withTranslation()(ArticlesPage);
