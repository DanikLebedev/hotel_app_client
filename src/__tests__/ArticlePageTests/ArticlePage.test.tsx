import { shallow } from 'enzyme';
import React from 'react';
import ArticlesPage from '../../pages/ArticlesPage/ArticlesPage';
import ArticlesList from '../../components/ArticlesList/ArticlesList';
import ArticleItem from '../../components/ArticleItem/ArticleItem';

const articleData = {
    title: '',
    image: '',
    text: '',
    createdAt: '',
};

describe('articles page test tests', () => {
    it('should renders room page without crashing', function() {
        const wrapper = shallow(<ArticlesPage />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should renders articles list without crashing', function() {
        const wrapper = shallow(<ArticlesList />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should renders articles item without crashing', function() {
        const wrapper = shallow(<ArticleItem articleInfo={articleData} />);
        expect(wrapper).toMatchSnapshot();
    });
});
