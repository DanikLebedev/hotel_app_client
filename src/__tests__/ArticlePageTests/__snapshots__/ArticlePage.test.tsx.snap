// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articles page test tests should renders articles item without crashing 1`] = `
<Col
  className="mb-4"
  lg={4}
  md={4}
  sm={12}
  xs={12}
>
  <img
    alt="img"
    className="articles-page__item-img"
    src="http://localhost:5000/static/"
  />
  <h3 />
  <div
    className="rooms-page__item-text"
  >
    <p />
    <p />
  </div>
  <NavLink
    className="button btn-black"
    id="redirect-button"
    to="/articles/undefined"
  >
    Read More
  </NavLink>
</Col>
`;

exports[`articles page test tests should renders articles list without crashing 1`] = `
<ForwardRef
  noGutters={false}
>
  <ArticleItem
    articleInfo={
      Object {
        "createdAt": "",
        "image": "",
        "text": "",
        "title": "",
      }
    }
    key="0"
  />
</ForwardRef>
`;

exports[`articles page test tests should renders room page without crashing 1`] = `
<ArticlesPage
  i18n={Object {}}
  t={[Function]}
  tReady={false}
/>
`;
